File: win-prog/ProgWinCS/Tree View and List View/SysInfoListView/SysInfoListView.cs
//----------------------------------------------
// SysInfoListView.cs © 2001 by Charles Petzold
//----------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class SysInfoListView: Form
{
     public static void Main()
     {
          Application.Run(new SysInfoListView());
     }
     public SysInfoListView()
     {
          Text = "System Information (List View)";

               // Create ListView control.

          ListView listview = new ListView();
          listview.Parent = this;
          listview.Dock = DockStyle.Fill;
          listview.View = View.Details;
          
               // Define columns based on maximum string widths.

          Graphics grfx = CreateGraphics();

          listview.Columns.Add("Property", 
                    (int) SysInfoReflectionStrings.MaxLabelWidth(grfx, Font),
                    HorizontalAlignment.Left);

          listview.Columns.Add("Value",
                    (int) SysInfoReflectionStrings.MaxValueWidth(grfx, Font),
                    HorizontalAlignment.Left);

          grfx.Dispose();

               // Get the data that will be displayed.

          int      iNumItems  = SysInfoReflectionStrings.Count;
          string[] astrLabels = SysInfoReflectionStrings.Labels;
          string[] astrValues = SysInfoReflectionStrings.Values;

               // Define the items and subitems.

          for (int i = 0; i < iNumItems; i++)
          {
               ListViewItem lvi = new ListViewItem(astrLabels[i]);
               lvi.SubItems.Add(astrValues[i]);
               listview.Items.Add(lvi);
          }
     }
}        

File: win-prog/ProgWinVB/Tree View and List View/SysInfoListView/SysInfoListView.vb
'------------------------------------------------
' SysInfoListView.vb (c) 2002 by Charles Petzold
'------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class SysInfoListView
    Inherits Form

    Shared Sub Main()
        Application.Run(New SysInfoListView())
    End Sub

    Sub New()
        Text = "System Information (List View)"

        ' Create ListView control.
        Dim listvu As New ListView()
        listvu.Parent = Me
        listvu.Dock = DockStyle.Fill
        listvu.View = View.Details

        ' Define columns based on maximum string widths.
        Dim grfx As Graphics = CreateGraphics()
        listvu.Columns.Add("Property", _
                CInt(SysInfoReflectionStrings.MaxLabelWidth(grfx, Font)), _
                HorizontalAlignment.Left)
        listvu.Columns.Add("Value", _
                CInt(SysInfoReflectionStrings.MaxValueWidth(grfx, Font)), _
                HorizontalAlignment.Left)
        grfx.Dispose()

        ' Get the data that will be displayed.
        Dim iNumItems As Integer = SysInfoReflectionStrings.Count
        Dim astrLabels() As String = SysInfoReflectionStrings.Labels
        Dim astrValues() As String = SysInfoReflectionStrings.Values

        ' Define the items and subitems.
        Dim i As Integer
        For i = 0 To iNumItems - 1
            Dim lvi As New ListViewItem(astrLabels(i))
            lvi.SubItems.Add(astrValues(i))
            listvu.Items.Add(lvi)
        Next i
    End Sub
End Class
